sljit_src/sljitProtExecAllocator.c

```diff
@@ -109,6 +109,13 @@ static SLJIT_INLINE int create_tempfile(void)
 	char *dir;
 	size_t len;
 
+#ifdef HAVE_MEMFD_CREATE
+	/* this is a GNU extension, make sure to use -D_GNU_SOURCE */
+	fd = memfd_create("sljit", MFD_CLOEXEC);
+	if (fd != -1)
+		return fd;
+#endif
```

Comment:
Shouldn't we return regardless of the result? If we continue the execution we just create another tmp file.

Reply:
if fd == -1, then no file was created and then we continue with the fallback to try with open and if that fails as well once more with mkostemp.

note the same logic was used for the test with open(O_TMPFILE)